os-appium-tests/tests/specs/contact.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+After each test scenario (or each suite of tests), the test should have a teardown where all contacts created are deleted
+However, this one doesn't have teardown, because the plugin can't delete contacts and, with appium, we can't access the native contacts and delete them
+*/
+
+// Import constants and classes needed
+import 'jasmine';
+import {DEFAULT_TIMEOUT, DEFAULT_TIMEOUT_INTERVAL} from '../constants';
+import * as Context from '../helpers/Context';
+import NativeAlert from '../helpers/NativeAlert';
+import * as ContactsScreen from '../screenobjects/ContactsScreen';
+
+describe('[TestSuite, Description("Add Contact and find it")]', () => {
+
+    // One contact created - it's used for all tests
+    beforeAll(() => {
+
+        // Switch the context to WEBVIEW
+        Context.switchToContext(Context.CONTEXT_REF.WEBVIEW);
+
+        // Enter Add Contact Scren
+        ContactsScreen.getAddContactScreen().click();
+        ContactsScreen.getTitle().waitForDisplayed(DEFAULT_TIMEOUT);
+
+        // Setup of the test
+        ContactsScreen.SetupContactAllParameters().click();
+
+        // Test: click to create the contact
+        ContactsScreen.getAddContactButton().click();
+
+        // In case an alert message appears to allow permissions to the phone, it clicks ALLOW
+        Context.switchToContext(Context.CONTEXT_REF.NATIVE);
+        if (NativeAlert.isShown(true, browser)) {
+            NativeAlert.waitForIsShown(true, browser);
+            NativeAlert.pressButton('ALLOW', browser);
+            NativeAlert.waitForIsShown(false, browser);
+            }
+        Context.switchToContext(Context.CONTEXT_REF.WEBVIEW);
+
+    });
+
+    it('[Test, Description("Add contact with all parameters"), Priority="P0"]', () => {
+
+        // The expected result is for the contact to be created (message text = true)
+        expect(ContactsScreen.getFeedbackMessage().getText()).toEqual('Success: True');
+
+        // discover a way to validate the number in the contacts
+    });
+
+    it('[Test, Description("Add contact with same number"), Priority="P2"]', () => {
+
+        // Setup of the test
+        ContactsScreen.SetupContactDifferentPhone().click();
+
+        // Test: click to create the contact
+        ContactsScreen.getAddContactButton().click();
+
+        // The expected result is for the contact to be created (message text = true)
```

Comment:
hummm... are we validating what the description is mention? "with the same number", can't we validate that?

Reply:
Yes!! You're right, in the next test case (Find Contact by First Name) I will add a validation to see if the new phone number was added to the contact. What do you think?
PS: the name of the test case was wrong, it should be: "Add contact with different number"
![image](https://user-images.githubusercontent.com/47888982/62611787-2a285100-b8fe-11e9-8bec-1b3ad10c2c1a.png)